toast-tk-runtime/src/main/java/io/toast/tk/runtime/utils/ArgumentHelper.java

```diff
@@ -137,14 +142,30 @@ private static String handleValueWithNestedVars(
 		final IActionItemRepository repoSetup,
 		String value
 	) {
-		final Matcher m = ACTION_ITEM_VAR_PATTERN.matcher(value);
+		// First we check Vars with parentesis 
+		value = handleValueWithNestedVars(repoSetup, ACTION_ITEM_VAR_PATTERN_1, value);
+		// Then without
+		return handleValueWithNestedVars(repoSetup, ACTION_ITEM_VAR_PATTERN_2, value);
+	}
+	private static String handleValueWithNestedVars(
+		final IActionItemRepository repoSetup, Pattern pattern,
+		String value
+	) {
+		final Matcher m = pattern.matcher(value);
 		int pos = 0;
 		while(m.find()) {
-			final String varName = m.group(pos + 1);
+			String varName = m.group(pos + 1);
+			if(pattern.equals(ACTION_ITEM_VAR_PATTERN_1)) {
+				varName = varName.substring(1, varName.length() - 1);
+			}
 			if(repoSetup.getUserVariables().containsKey(varName)) {
-				final Object varValue = repoSetup.getUserVariables().get(varName);
+				String varValue = repoSetup.getUserVariables().get(varName).toString();
 				if(!varName.equals(varValue)){
-					value = value.replaceFirst("\\" + varName + "\\b", (String) varValue);
+					String valueToReplace = "\\" + varName + "\\b";
+					if(pattern.equals(ACTION_ITEM_VAR_PATTERN_1)) {
+						valueToReplace = "\\(" + valueToReplace + "\\)";
+					}
+					value = value.replaceFirst(valueToReplace, varValue.replace("$", "\\$"));	
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "value". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)